pkg/dockerregistry/server/quotarestrictedblobstore.go

```diff
@@ -35,16 +35,57 @@ import (
 	"fmt"
 	"github.com/docker/distribution"
 	"github.com/docker/distribution/context"
+	"os"
+	"strconv"
 	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
 
 	kapi "k8s.io/kubernetes/pkg/api"
-	kerrors "k8s.io/kubernetes/pkg/api/errors"
 	"k8s.io/kubernetes/pkg/api/resource"
 	"k8s.io/kubernetes/pkg/quota"
 
+	imageadmission "github.com/openshift/origin/pkg/image/admission"
 	imageapi "github.com/openshift/origin/pkg/image/api"
 )
 
+const (
+	projectCacheTTLEnvVar  = "DOCKER_REGISTRY_PROJECT_CACHE_TTL"
+	defaultProjectCacheTTL = time.Minute
+)
+
+var (
+	// projectResourceQuotas is a shared cache of resource quota objects keyed by project name.
+	// Thread safe and reused by all middleware layers.
+	projectResourceQuotas projectObjectListStore
+	// projectLimitRanges is a shared cache of limit range objects keyed by project name.
+	// Thread safe and reused by all middleware layers.
+	projectLimitRanges projectObjectListStore
+)
+
+func init() {
+	if os.Getenv(enforceQuotaEnvVar) != "true" {
```

Comment:
It is disabling the feature. See [Blobs()](https://github.com/openshift/origin/pull/8195/files/009f353ce0f5825f8e3821be026ae9f687ab66dc#diff-1a18cb8c0d7684cefcef809b9d01aac3R169). This just skips unnecessary module initialization.